### PR TITLE
Where method and where function

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented `cupy.where` partially, and `Where` function for chainer.

### CODE REVIEW

File: chainer/functions/array/where.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    def forward(self, inputs):
+        xp = cuda.get_array_module(*inputs)
+        condition, x, y = inputs
+        return xp.where(condition, x, y),
+
+    def backward(self, inputs, grads):
+        xp = cuda.get_array_module(*inputs)
+        condition, x, y = inputs
+        gx = xp.where(condition, grads[0], 0)
+        gy = xp.where(~condition, grads[0], 0)
```
Comment (okuta) [inline chainer/functions/array/where.py]: `gy = xp.where(condition, 0, grads[0])` is better.

File: cupy/sorting/search.py [hunk 2]
```diff
@@ -60,7 +61,40 @@ def argmin(a, axis=None, dtype=None, out=None, keepdims=False):
 # TODO(okuta): Implement flatnonzero
 
 
-# TODO(okuta): Implement where
+def where(condition, x=None, y=None):
+    """Return elements, either from x or y, depending on condition.
+
+    .. note::
+
+       Currently Cupy doesn't support ``wherer(condition)``, that Numpy
```
Comment (okuta) [inline cupy/sorting/search.py]: typo

File: cupy/sorting/search.py [hunk 3]
```diff
@@ -60,7 +61,40 @@ def argmin(a, axis=None, dtype=None, out=None, keepdims=False):
...
+
+    missing = (x is None, y is None).count(True)
+
+    if missing == 1:
+        raise ValueError("Must provide both 'x' and 'y' or neither.")
+    if missing == 2:
+        # TODO(unno): return nonzero(cond)
+        return NotImplementedError()
+
+    return _where_ufunc(condition, x, y)
```
Comment (okuta) [inline cupy/sorting/search.py]: `return _where_ufunc(condition.astype('?'), x, y)` is better.

File: chainer/functions/array/where.py (hunk 1 above)
Comment (unnonouno) [inline chainer/functions/array/where.py]: !

**Comment by okuta** [comment]
LGTM!

**Pull request merged**